packages/evm/src/stack.ts

```diff
@@ -9,38 +7,32 @@ export class Stack {
   _store: bigint[]
   _maxHeight: number
 
+  private _len: number = 0
+
   constructor(maxHeight?: number) {
-    this._store = []
+    this._store = [] //Array(maxHeight ?? 1024) -> initializing the array makes the constructor 10x slower
```

Comment:
Stack only gets initialized on creating a `new Interpreter`. This only happens if you enter a new "call frame", so either: the transition from tx -> calling / creating a contract, or using opcodes CALL, CALLCODE, DELEGATECALL, STATICCALL, CREATE, CREATE2. (In practice this thus does not happen a lot since there is a rather big base gas cost for these opcodes)

Reply:
Note: with TransientStorage, each call frame gets a extra stack (so for Cancun the amount of times the constructor gets called doubles when compared <= Cancun)